### PR TITLE
Misc fixes && improvements 

### PR DESCRIPTION
**Pull request by sferhah**

### CODE REVIEW

**Comment by crobibero** [comment]
I apprecate all of the work youre doing but I highly suggest splitting this into seperate PRs so I can review easier

If that is too much work for this just keep it in mind for potential future PRs

**Comment by sferhah** [comment]
Ok. I'll try to do that during the week-end.

File: src/Jellyfin.Maui/Models/ServerStateModel.cs [hunk 1]
```diff
@@ -1,35 +1,22 @@
...
-    /// <param name="id">The server id.</param>
-    /// <param name="name">The server name.</param>
-    /// <param name="url">The server url.</param>
-    public ServerStateModel(Guid id, string name, string url)
-    {
-        Id = id;
-        Name = name;
-        Url = url;
-    }
+    required public Guid Id { get; init; }
```
Comment (crobibero) [inline src/Jellyfin.Maui/Models/ServerStateModel.cs]:
```suggestion
    public required Guid Id { get; init; }
```

Fix ordering

File: src/Jellyfin.Maui/Models/ServerStateModel.cs [hunk 2]
```diff
@@ -1,35 +1,22 @@
...
 
     /// <summary>
-    /// Gets the server id.
+    /// Gets or sets the server name.
     /// </summary>
-    public Guid Id { get; init; }
+    required public string Name { get; set; }
```
Comment (crobibero) [inline src/Jellyfin.Maui/Models/ServerStateModel.cs]:
```suggestion
    public required string Name { get; set; }
```

File: src/Jellyfin.Maui/Models/ServerStateModel.cs [hunk 3]
```diff
@@ -1,35 +1,22 @@
...
-    /// Gets the server name.
+    /// Gets or sets the server url.
     /// </summary>
-    public string Name { get; init; }
-
-    /// <summary>
-    /// Gets the server url.
-    /// </summary>
-    public string Url { get; init; }
+    required public string Url { get; set; }
```
Comment (crobibero) [inline src/Jellyfin.Maui/Models/ServerStateModel.cs]:
```suggestion
    public required string Url { get; set; }
```

File: src/Jellyfin.Maui/Models/UserStateModel.cs [hunk 4]
```diff
@@ -6,37 +6,22 @@
...
-    /// <param name="name">The user name.</param>
-    /// <param name="token">The user token.</param>
-    public UserStateModel(Guid id, Guid serverId, string name, string token)
-    {
-        Id = id;
-        ServerId = serverId;
-        Name = name;
-        Token = token;
-    }
+    required public Guid Id { get; init; }
```
Comment (crobibero) [inline src/Jellyfin.Maui/Models/UserStateModel.cs]:
```suggestion
    public required Guid Id { get; init; }
```

File: src/Jellyfin.Maui/Models/UserStateModel.cs [hunk 5]
```diff
@@ -6,37 +6,22 @@
...
 
     /// <summary>
-    /// Gets the user id.
+    /// Gets or sets the server id the user belongs to.
     /// </summary>
-    public Guid Id { get; init; }
+    required public Guid ServerId { get; init; }
```
Comment (crobibero) [inline src/Jellyfin.Maui/Models/UserStateModel.cs]:
```suggestion
    public required Guid ServerId { get; init; }
```

File: src/Jellyfin.Maui/Models/UserStateModel.cs [hunk 6]
```diff
@@ -6,37 +6,22 @@
...
-    /// Gets the user name.
+    /// Gets or sets the user token.
     /// </summary>
-    public string Name { get; init; }
-
-    /// <summary>
-    /// Gets the user token.
-    /// </summary>
-    public string Token { get; init; }
+    required public string Token { get; set; }
```
Comment (crobibero) [inline src/Jellyfin.Maui/Models/UserStateModel.cs]:
```suggestion
    public required string Token { get; set; }
```

File: src/Jellyfin.Maui/Pages/LoadingPage.xaml.cs [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="LoadingPage"/> class.
+    /// </summary>
+    public LoadingPage()
+    {
+        InitializeComponent();
+    }
+
+    protected override async void OnAppearing()
```
Comment (crobibero) [inline src/Jellyfin.Maui/Pages/LoadingPage.xaml.cs:21]:
```suggestion
    /// <inheritdoc/>
    protected override async void OnAppearing()
```

File: src/Jellyfin.Maui/Models/UserStateModel.cs [hunk 8]
```diff
@@ -6,37 +6,22 @@
...
-        Id = id;
-        ServerId = serverId;
-        Name = name;
-        Token = token;
-    }
+    required public Guid Id { get; init; }
 
     /// <summary>
-    /// Gets the user id.
+    /// Gets or sets the server id the user belongs to.
```
Comment (crobibero) [inline src/Jellyfin.Maui/Models/UserStateModel.cs]:
```suggestion
    /// Gets the server id the user belongs to.
```

File: src/Jellyfin.Maui/Models/UserStateModel.cs [hunk 9]
```diff
@@ -6,37 +6,22 @@
 public class UserStateModel
 {
     /// <summary>
-    /// Initializes a new instance of the <see cref="UserStateModel"/> class.
+    /// Gets or sets the user id.
```
Comment (crobibero) [inline src/Jellyfin.Maui/Models/UserStateModel.cs]:
```suggestion
    /// Gets the user id.
```

`init` setters should not be part of the documentation

File: src/Jellyfin.Maui/Services/StateStorageService.cs [hunk 10]
```diff
@@ -121,41 +112,38 @@ public async ValueTask AddUserAsync(UserStateModel userStateModel)
...
-                    }
-                }
-
-                state.Servers.RemoveAt(index);
-                break;
-            }
+            return;
+        }
+
+        foreach (var user in state.Users.Where(x => x.ServerId == serverId).ToList())
```
Comment (crobibero) [inline src/Jellyfin.Maui/Services/StateStorageService.cs]:
```suggestion
        foreach (var user in state.Users.Where(x => x.ServerId == serverId))
```

`.ToList()` shouldn't be required here

File: src/Jellyfin.Maui/ViewModels/Login/LoginViewModel.cs [hunk 11]
```diff
@@ -93,9 +96,20 @@ public override async ValueTask InitializeAsync()
...
+        Loading = false;
+
+        CheckQuickConnectAvailablity().SafeFireAndForget();
+    }
 
+    private async Task CheckQuickConnectAvailablity()
+    {
+        CheckingQuickConnectAvailablity = true;
+
+        await Task.Delay(2_000).ConfigureAwait(false);  // mock slow network
```
Comment (crobibero) [inline src/Jellyfin.Maui/ViewModels/Login/LoginViewModel.cs:109]: The mocked delay should only occur on a debug build

**Review by crobibero — CHANGES_REQUESTED** [review]

**Comment by crobibero** [comment]
Closing in favor of https://github.com/crobibero/jellyfin-client-maui/pull/65

**Pull request closed without merging**